lib/mayaHydra/hydraExtensions/adapters/materialNetworkConverter.cpp

```diff
@@ -395,6 +395,74 @@ class MayaHydraUvAttrConverter : public MayaHydraMaterialAttrConverter
 }; // namespace
 
+class MayaHydraMixedColorMaterialAttrConverter : public MayaHydraComputedMaterialAttrConverter
+{
+public:
+    MayaHydraMixedColorMaterialAttrConverter(
+        const TfToken&          remappedBaseColorName,
+        const TfToken&          remappedBaseWeightName,
+        const TfToken&          remappedCoatColorName,
+        const TfToken&          remappedCoatWeightName)
+        : _remappedBaseColorName(remappedBaseColorName)
+        , _remappedBaseWeightName(remappedBaseWeightName)
+        , _remappedCoatColorName(remappedCoatColorName)
+        , _remappedCoatWeightName(remappedCoatWeightName)
+    {
+    }
+
+    SdfValueTypeName GetType() override { return SdfValueTypeNames->Vector3f; }
+
+    VtValue GetValue(
+        MFnDependencyNode&      node,
+        const TfToken&          paramName,
+        const SdfValueTypeName& type,
+        const VtValue*          fallback = nullptr,
+        MPlugArray*             outPlug = nullptr) override
+    {
+        VtValue baseColor = MayaHydraMaterialNetworkConverter::ConvertMayaAttrToScaledValue(
+            node,
+            _remappedBaseColorName.GetText(),
+            _remappedBaseWeightName.GetText(),
+            SdfValueTypeNames->Vector3f,
+            fallback,
+            outPlug);
+
+        // Mix baseColor with coat coatColor
+        if (baseColor.IsHolding<GfVec3f>()) {
+            VtValue coatWeight = MayaHydraMaterialNetworkConverter::ConvertMayaAttrToValue(
+                node,
+                _remappedCoatWeightName.GetText(),
+                SdfValueTypeNames->Float,
+                fallback,
+                outPlug);
+            if (coatWeight.IsHolding<float>()) {
+                float coatWeightFloat = coatWeight.UncheckedGet<float>();
+                if (coatWeightFloat != 0.0f) {
+                    VtValue coatColor = MayaHydraMaterialNetworkConverter::ConvertMayaAttrToValue(
+                        node,
+                        _remappedCoatColorName.GetText(),
+                        SdfValueTypeNames->Vector3f,
+                        fallback,
+                        outPlug);
+                    if (coatColor.IsHolding<GfVec3f>()) {
+                        GfVec3f baseColorVec3f = baseColor.UncheckedGet<GfVec3f>();
+                        GfVec3f coatColorVec3f = coatColor.UncheckedGet<GfVec3f>();
+                        GfVec3f coatAttenuationVec3f = GfLerp(coatWeightFloat, GfVec3f(1.0f, 1.0f, 1.0f), coatColorVec3f);
+                        return VtValue(GfCompMult(baseColorVec3f, coatAttenuationVec3f));
+                    }
+                }
+            }
+        }
+
+        return baseColor;
+    }
+
+ protected:
+    const TfToken& _remappedBaseColorName;
+    const TfToken& _remappedBaseWeightName;
+    const TfToken& _remappedCoatColorName;
+    const TfToken& _remappedCoatWeightName;
+};
 
 class MayaHydraOpenPBREmissionColorMaterialAttrConverter : public MayaHydraComputedMaterialAttrConverter
```

Comment:
This will be eventually replaced by MayaHydraMixedColorMaterialAttrConverter when Maya has emissionWeight exposed.